This patch lets po/Makefile find mkinstalldirs from the aux directory
even if $srcdir isn't absolute.


To apply this in the gettext sources:

# cd /usr/local/src/gettext-0.10.35
(or wherever you have extracted the source tree)
# patch -p1 </PATH/TO/gettext-0.10.35-6-mkinstalldirs.diff
# aclocal
# autoconf

After installing the patched gettext, rerun `gettextize -f', `aclocal'
and `autoconf' in your favorite packages to let them use the changes.


diff -Nurp gettext-0.10.35.kn.5/m4/ChangeLog gettext-0.10.35.kn.6/m4/ChangeLog
--- gettext-0.10.35.kn.5/m4/ChangeLog	Wed Sep 23 21:38:12 1998
+++ gettext-0.10.35.kn.6/m4/ChangeLog	Fri Dec 18 17:52:44 1998
@@ -1,3 +1,9 @@
+1998-12-18  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* gettext.m4 (AM_GNU_GETTEXT): Cope with absolute $ac_aux_dir.
+	Format MKINSTALLDIRS so that Makefiles don't have to prepend
+	anything to it.
+
 1998-09-23  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
 
 	* gettext.m4 (AM_WITH_NLS): Set gt_cv_func_gettext_libintl=no if
diff -Nurp gettext-0.10.35.kn.5/m4/gettext.m4 gettext-0.10.35.kn.6/m4/gettext.m4
--- gettext-0.10.35.kn.5/m4/gettext.m4	Wed Sep 23 21:38:12 1998
+++ gettext-0.10.35.kn.6/m4/gettext.m4	Fri Dec 18 17:59:40 1998
@@ -292,11 +292,19 @@ strcasecmp strdup __argz_count __argz_st
    AC_SUBST(GT_YES)
 
    dnl If the AC_CONFIG_AUX_DIR macro for autoconf is used we possibly
-   dnl find the mkinstalldirs script in another subdir but ($top_srcdir).
-   dnl Try to locate is.
+   dnl find the mkinstalldirs script in another subdir but $(top_srcdir).
+   dnl Try to locate it.
+   dnl Format @MKINSTALLDIRS@ so that it can be used in Makefiles of all
+   dnl directories.  This may involve referring to make variables.
    MKINSTALLDIRS=
    if test -n "$ac_aux_dir"; then
-     MKINSTALLDIRS="$ac_aux_dir/mkinstalldirs"
+     dnl In Autoconf 2.12, $ac_aux_dir can be either absolute
+     dnl (if $srcdir was absolute) or relative to the top build directory.
+     if echo "$ac_aux_dir" | grep "^/" >/dev/null; then
+       MKINSTALLDIRS="$ac_aux_dir/mkinstalldirs"
+     else
+       MKINSTALLDIRS="\$(top_builddir)/$ac_aux_dir/mkinstalldirs"
+     fi
    fi
    if test -z "$MKINSTALLDIRS"; then
      MKINSTALLDIRS="\$(top_srcdir)/mkinstalldirs"
diff -Nurp gettext-0.10.35.kn.5/po/ChangeLog gettext-0.10.35.kn.6/po/ChangeLog
--- gettext-0.10.35.kn.5/po/ChangeLog	Fri Dec 18 17:53:20 1998
+++ gettext-0.10.35.kn.6/po/ChangeLog	Fri Dec 18 17:44:37 1998
@@ -1,3 +1,9 @@
+1998-12-18  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
+
+	* Makefile.in.in (top_builddir): New variable.
+	(MKINSTALLDIRS): Don't prepend $(top_srcdir); the configure script
+	takes care of it.
+
 1998-11-05  Kalle Olavi Niemitalo  <tosi@stekt.oulu.fi>
 
 	* Makefile.in.in (DESTDIR): New variable.  It's empty unless
diff -Nurp gettext-0.10.35.kn.5/po/Makefile.in.in gettext-0.10.35.kn.6/po/Makefile.in.in
--- gettext-0.10.35.kn.5/po/Makefile.in.in	Fri Dec 18 17:11:47 1998
+++ gettext-0.10.35.kn.6/po/Makefile.in.in	Fri Dec 18 23:50:23 1998
@@ -1,7 +1,7 @@
 # Makefile for program source directory in GNU NLS utilities package.
 # Copyright (C) 1995, 1996, 1997 by Ulrich Drepper <drepper@gnu.ai.mit.edu>
 #
-# This file file be copied and used freely without restrictions.  It can
+# This file can be copied and used freely without restrictions.  It can
 # be used in projects which are not available under the GNU Public License
 # but which still want to provide support for the GNU gettext functionality.
 # Please note that the actual code is *not* freely available.
@@ -23,12 +23,13 @@ localedir = $(datadir)/locale
 gnulocaledir = $(prefix)/share/locale
 gettextsrcdir = $(prefix)/share/gettext/po
 subdir = po
+top_builddir = ..
 
 DESTDIR =
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = $(top_srcdir)/@MKINSTALLDIRS@
+MKINSTALLDIRS = @MKINSTALLDIRS@
 
 CC = @CC@
 GENCAT = @GENCAT@
